pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -84,7 +84,7 @@ func alterChangefeedPlanHook(
 	}
 
 	fn := func(ctx context.Context, _ []sql.PlanNode, resultsCh chan<- tree.Datums) error {
-		typedExpr, err := alterChangefeedStmt.Jobs.TypeCheck(ctx, p.SemaCtx(), types.Int)
+		typedExpr, err := tree.TypeCheckAndRequire(ctx, alterChangefeedStmt.Jobs, p.SemaCtx(), types.Int, "get-job-id")
```

Comment:
i don't know the full history of why `alterChangefeedStmt.Jobs` is an expression, but looking a bit closer, i'm not sure this will work either. For example: `alter changefeed (select 1) add b;` also crashes the node.

what we'd need here is to evaluate the `alterChangefeedStmt.Jobs` expression. or else change `alterChangefeedStmt.Jobs` so it is not an expression.